jena-base/src/main/java/org/apache/jena/atlas/iterator/Iter.java

```diff
@@ -29,6 +29,14 @@
 import org.apache.jena.atlas.lib.Closeable ;
 import org.apache.jena.atlas.lib.Sink ;
 
+/**
+ * Iter provides utilities for working with {@link Iterator}s.
+ *
+ * Iter should never be used as a return type or parameter type in the public contract of a class. It is only to be used
+ * inside implementation code and is instantiated only to allow method-chaining as part of a calculation.
```

Comment:
What I meant by "difference" was context of use. In other words, what I have been trying to get at with this conversation, as I have written, is the answer to "When should one use `Iter`, and when should one use `ExtendedIterator`?" I tried to use `Iter` inside an `ExtendedIterator` during JENA-966 and you explained that this was wrong, and I am trying to get Javadocs into these classes so that other people don't have the same confusion. 

I still have no idea when to use either. If the difference is purely historical, then there is no inherent need for both, and perhaps I should open a ticket for a slow, careful, un-disruptive migration towards a single type for these purposes?